rust-src/crypto_common/src/serialize.rs

```diff
@@ -1,15 +1,14 @@
 pub use crate::impls::*;
 
-use core::cmp;
-
-use std::{convert::TryFrom, marker::PhantomData};
-
+use anyhow::bail;
 use byteorder::{BigEndian, ReadBytesExt, WriteBytesExt};
-use failure::Fallible;
-use std::collections::btree_map::BTreeMap;
+use core::cmp;
+use std::{collections::btree_map::BTreeMap, convert::TryFrom, marker::PhantomData};
 
 static MAX_PREALLOCATED_CAPACITY: usize = 4096;
 
+pub type ParseResult<T, E = anyhow::Error> = anyhow::Result<T, E>;
```

Comment:
I don't think this is a good idea. Either it should be just
```
ParseResult<T> = anyhow::Result<T>
```
or just use `Result` directly. That is how we use it elsewhere.